.github/composites/docker-build-push/action.yaml

```diff
@@ -0,0 +1,115 @@
+name: 'Build and Publish Docker Image'
+description: 'Build and Publish Docker Image'
+inputs:
+  repository:
+    description: 'Docker hub repository'
+    required: true
+    default: 'cloudcustodian'
+  name:
+    description: 'image name, should match dockerfile name'
+    required: true
+  push:
+    description: 'Push image: true/false'
+    required: true
+    default: false
+  poetry_version:
+    description: "Poetry Version to use"
+    default: "1.1.14"
+  trivy_version:
+    description: "trivy version to use"
+    default: "0.5.4"
+  platforms:
+    description: "Platforms to build, e.g. linux/arm64,linux/amd64"
+    default: "linux/amd64"
+runs:
+  using: "composite"
+  steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+
+    # until we get read only credentials, we'll only login if we're pushing
+    # note that we can hit rate limits on pulling though
+    - name: Login to Docker Hub
+      if: "${{ inputs.push == 'true' }}"
+      uses: docker/login-action@v1
+      with:
+        username: ${{ env.HUB_USER }}
+        password: ${{ env.HUB_TOKEN }}
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: |
+          ${{ inputs.repository }}/${{ inputs.name }}
+        tags: |
+          type=semver,pattern={raw}
+          type=raw,value=latest
+        sep-tags: ','
+
+    # build a single platform image first in order to actually load the image from buildx
+    # manifests are not supported yet
+    # https://github.com/docker/buildx/issues/166
+    # https://github.com/docker/buildx/issues/59
+    - name: Build
+      uses: docker/build-push-action@v3.1.1
+      with:
+        context: .
+        build-args: |
+          POETRY_VERSION=${{ inputs.poetry_version }}
+        push: false
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+        # makes the image available in docker, e.g. docker images inspect
+        load: true
+        file: "docker/${{ inputs.name }}"
+
+    - name: scan
```

Comment:
nit: scan and test are the only lower case steps in the composite, would be nice to consistently case.